src/sentry/api/endpoints/release_thresholds/health_checks/is_new_issues_count_healthy.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import annotations
+
+from logging import getLogger
+
+from sentry.api.endpoints.release_thresholds.types import EnrichedThreshold
+from sentry.models.group import Group
+from sentry.models.groupenvironment import GroupEnvironment
+from sentry.models.release_threshold.constants import TriggerType
+
+logger = getLogger("sentry.releases.servicer")
+
+
+def is_new_issues_count_healthy(release_threshold: EnrichedThreshold) -> bool:
+    release_id = release_threshold["release_id"]
+    project_id = release_threshold["project_id"]
+    time_range = (release_threshold["start"], release_threshold["end"])
+    try:
+        if release_threshold["environment"]:
+            new_issues = GroupEnvironment.objects.filter(
```

Comment:
So actually
Rather than constructing 2 separate filters, why don't we construct a single query and we can add filters to it as needed

```
query = GroupEnvironment.objects.filter(
  first_seen__range=time_range,
  first_release__id__in=release_list,
  group__project__id__in=project_list,
)

if environments:
  query.filter(environment__id__in=environment_list)
```

Fun fact - Django doesn't actually execute the query until the query is evaluated, so we can do fancy stuff like this to construct our queries dynamically

Reply:
Oh - so `.count()` would be considered an `evaluation` and would actually trigger the query execution.

similarly, iterating through, converting to a list/set, etc. etc. are all executable events

Reply:
Because it's 2 different models, not the same model

Reply:
I see I see

So actually, we could still do the same, but prefetch the `GroupEnvironment`
Then if there's an environment, we could simply fitler on that again.

```
query = Group.objects.filter(
    project__id__in=project_id_list,
    release__version__in=release_versions_list,
)

if environments:
    query.filter(groupenvironment__environment__id__in=environments_id_list)
```

Same sort of deal since it's ultiamtely the groups that we're trying to count here